token/program-2022/src/extension/confidential_mint_burn/instruction.rs

```diff
@@ -0,0 +1,556 @@
+    ///      pre-verified `VerifyBatchedGroupedCiphertext3HandlesValidity` proof
+    ///   5. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyBatchedRangeProofU128`
+    ///   6. `[]` The multisig account owner.
+    ///   7.. `[signer]` Required M signer accounts for the SPL Token Multisig
+    ///
+    /// Data expected by this instruction:
+    ///   `MintInstructionData`
+    Mint,
+    /// Burn tokens from confidential balance
+    ///
+    /// Fails if the destination account is frozen.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single authority
+    ///   0. `[writable]` The SPL Token account.
+    ///   1. `[]` The SPL Token mint. `[writable]` if the mint has a non-zero
+    ///      supply elgamal-pubkey
+    ///   2. `[]` (Optional) Instructions sysvar if at least one of the
+    ///      `zk_elgamal_proof` instructions are included in the same
+    ///      transaction.
+    ///   3. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyCiphertextCommitmentEquality` proof
+    ///   4. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyBatchedGroupedCiphertext3HandlesValidity` proof
+    ///   5. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyBatchedRangeProofU128`
+    ///   6. `[signer]` The single account owner.
+    ///
+    ///   * Multisignature authority
+    ///   0. `[writable]` The SPL Token mint.
+    ///   1. `[]` The SPL Token mint. `[writable]` if the mint has a non-zero
+    ///      supply elgamal-pubkey
+    ///   2. `[]` (Optional) Instructions sysvar if at least one of the
+    ///      `zk_elgamal_proof` instructions are included in the same
+    ///      transaction.
+    ///   3. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyCiphertextCommitmentEquality` proof
+    ///   4. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyBatchedGroupedCiphertext3HandlesValidity` proof
+    ///   5. `[]` (Optional) The context state account containing the
+    ///      pre-verified `VerifyBatchedRangeProofU128`
+    ///   6. `[]` The multisig account owner.
+    ///   7.. `[signer]` Required M signer accounts for the SPL Token Multisig
+    ///
+    /// Data expected by this instruction:
+    ///   `BurnInstructionData`
+    Burn,
+}
+
+/// Data expected by `ConfidentialMintBurnInstruction::InitializeMint`
+#[cfg_attr(feature = "serde-traits", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde-traits", serde(rename_all = "camelCase"))]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable)]
+#[repr(C)]
+pub struct InitializeMintData {
+    /// Authority used to modify the `ConfidentialMintBurn` mint
+    /// configuration and mint new tokens
+    pub authority: OptionalNonZeroPubkey,
```

Comment:
nit: since it isn't actually possible to initialize with a `None` pubkey, can you add a comment saying that this is really ` Pubkey` but the type is used for cleaner code during initialization?